src/solvers/refinement/string_constraint_generator.h

```diff
@@ -363,18 +370,21 @@ class string_constraint_generatort: messaget
     exprt char1, exprt char2, exprt char_a, exprt char_A, exprt char_Z);
   bool is_constant_string(const string_exprt &expr) const;
   string_exprt empty_string(const refined_string_typet &ref_type);
+  unsigned long to_integer_or_default(const exprt &expr, unsigned long def);
```

Comment:
Because `mp_integer` has a `integer2ulong` method. Valid radixes should be between 2 and 36, but I wrote this function to be more general. I'm happy to change the return value if you like.